elyra/pipeline/validate.py

```diff
@@ -251,9 +252,15 @@ async def _validate_node_properties(self, pipeline: dict, pipeline_runtime: str,
             components = ComponentRegistry.to_canvas_palette(component_list, categories)
             for node in node_list:
                 if node['type'] == 'execution_node':
-                    if self._is_legacy_pipeline(pipeline) and not self._is_standalone_submission(pipeline_source):
+                    if self._is_standalone_submission(pipeline_source):
+                        node_data = node['app_data'].get('component_parameters')
+                        node_label = node['app_data'].get('label')
+                    elif self._is_legacy_pipeline(pipeline):
                         node_data = node['app_data']
-                        node_label = node_data['ui_data'].get('label')
+                        if node_data.get('ui_data'):  # If present, always the source of truth
```

Comment:
I guess I figured this block should be used unconditionally since it's always correct irrespective of submission style.  Then only `node_data` would vary (which is unfortunate - see next).  

To @lresende's point, would updating the "template" for standalone submission to build `['app_data']['component_parameters']` then allow all of this to be agnostic as to how a pipeline's content got here?

Reply:
I would be all for consistency in updating the template to align better with how full pipelines are being submitted